### PR TITLE
feat: Adding dynamic vpc block to handle private zones

### PR DESCRIPTION
**Pull request by hawkesnc**
## Description
Taken from https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_zone#private-zone

This PR allows you to add a `vpc_block` to specify which AWS vpc you want to attach this route53 zone to and create a private hosted zone

## Motivation and Context
This PR allows you to create a private hosted zone(s)

## Breaking Changes
No breaking changes, adding new functionality.

## How Has This Been Tested?
- Tested by creating one / many private hosted zones
- Tested previous functionality by created one / many public hosted zones
- Caution: A limitation of Terraform ("all map elements must have the same type") does not allow you to create a mix of private/public zones in a single `zones` variable.

### CODE REVIEW

**Comment by hawkesnc** [comment]
@antonbabenko Let me know if there is anything that I need to add / do. Thanks! 😄

**Comment by andormarkus** [comment]
Hi @hawkesnc

I'm missing the following to able to approve this PR.
 - documentation update
 - variables.tf update for this new conditional variable

Thanks

**Comment by hawkesnc** [comment]
Hi @andormarkus, thanks for taking a look at my PR 😄 

In terms of the documentation update, the `vpc` is a dynamic block and I noticed that other dynamic blocks aren't documented (eg. `alias` in `modules/records/main.tf:25`). How should I proceed for this?

I did update the example showing how to include the dynamic vpc block. Please let me know if that works!

Thanks

**Comment by andormarkus** [comment]
Hi @hawkesnc 

It would be great if document  is a dynamic block and I noticed that other dynamic blocks aren't documented (eg. `alias` in `modules/records/main.tf:25`)

Please can you add these blocks to `README.md` as well.
Many developers are lazy and they don't looks further into the repo.


Thanks

**Comment by l13t** [comment]
@hawkesnc i think it would be nice if you add into example usage of module for both zones types: private and public.

P.S.
tested it for our infra and it works fine for us.

**Comment by badsector3** [comment]
Doesn't work for me. 

```
  on modules/route53/modules/zones/main.tf line 19, in resource "aws_route53_zone" "this":
  19:     for_each = lookup(each.value, "vpc_id", {})
    |----------------
    | each.value is object with 3 attributes

Cannot use a string value in for_each. An iterable collection is required.
```

That works  though:
```
  dynamic "vpc" {
    for_each  = each.value["vpc_id"] != null ? [{
      vpc_id  = each.value["vpc_id"]
    }] : []

    content {
      vpc_id  = vpc.value["vpc_id"]
    }
  }
```

Only problem is  that you have to specify vpc_id = null if public zone.
BTW Terraform 0.13.05

File: modules/zones/main.tf [hunk 1]
```diff
@@ -5,5 +5,12 @@ resource "aws_route53_zone" "this" {
   comment       = lookup(each.value, "comment", null)
   force_destroy = lookup(each.value, "force_destroy", false)
 
+  dynamic "vpc" {
+    for_each = lookup(each.value, "vpc", {})
+    content {
+      vpc_id = vpc.value
```
Comment (antonbabenko) [inline modules/zones/main.tf:11]: Could you also add support for the optional `vpc_region` as described in the docs [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_zone)?

**Comment by antonbabenko** [comment]
[v1.4.0](https://github.com/terraform-aws-modules/terraform-aws-route53/releases/v1.4.0) has been just released.

**Pull request merged**